project-api-tests/src/test/java/org/ballerina/projectapi/MavenCustomRepoTest.java

```diff
@@ -0,0 +1,205 @@
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.createSettingToml;
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.deleteArtifacts;
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.deleteFiles;
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.getEnvVariables;
+import static org.ballerina.projectapi.MavenCustomRepoTestUtils.getString;
+import static org.ballerina.projectapi.TestUtils.DISTRIBUTION_FILE_NAME;
+import static org.ballerina.projectapi.TestUtils.OUTPUT_CONTAIN_ERRORS;
+import static org.ballerina.projectapi.TestUtils.executeBuildCommand;
+import static org.ballerina.projectapi.TestUtils.executePullCommand;
+import static org.ballerina.projectapi.TestUtils.executePushCommand;
+
+/**
+ * Tests related to Maven repositories.
+ */
+public class MavenCustomRepoTest {
+
+    private static final String org = "luheerathan";
+    private static final String packagename = "pact";
+    private static final String version = "0.2.0";
+    private static final String GITHUB_REPO_ID = "github1";
+    private Path actualHomeDirectory;
+    private Path tempWorkspaceDirectory;
+    private Path actualHomeDirectoryClone;
+    private Map<String, String> envVariables;
+    @BeforeClass()
+    public void setUp() throws IOException {
+        TestUtils.setupDistributions();
+        actualHomeDirectory = Paths.get(System.getProperty("user.home")).resolve(".ballerina");
+        actualHomeDirectoryClone = Files.createTempDirectory("bal-test-integration-packaging-home-")
+                .resolve(".ballerina");
+        Files.walkFileTree(actualHomeDirectory,
+                new MavenCustomRepoTest.Copy(actualHomeDirectory, actualHomeDirectoryClone));
+        deleteFiles(actualHomeDirectory, true);
+        tempWorkspaceDirectory = Files.createTempDirectory("bal-test-integration-packaging-workspace-");
+
+        createSettingToml(actualHomeDirectory);
+        System.setProperty("user.home", actualHomeDirectory.getParent().toString());
+        envVariables = getEnvVariables();
+
+        // Copy test resources to temp workspace directory
+        try {
+            URI testResourcesURI = Objects.requireNonNull(getClass().getClassLoader()
+                    .getResource("maven-repos")).toURI();
+            Files.walkFileTree(Paths.get(testResourcesURI), new MavenCustomRepoTest.Copy(Paths.get(testResourcesURI),
+                    this.tempWorkspaceDirectory));
+        } catch (URISyntaxException e) {
+            Assert.fail("error loading resources");
+        }
+    }
+
+    @Test(description = "Push package to Github packages")
+    public void testPushBalaGithub() throws IOException, InterruptedException {
+        List<String> args = new ArrayList<>();
```

Comment:
What will happen when this test in the 2nd run? Since we are using a fixed name, won't it throw an exception when trying to push the bala again?

Reply:
No.. We are removing the bala once all the tests are done.. https://github.com/ballerina-platform/ballerina-distribution/pull/4810/files#diff-d5006a3d741d32d4cd70291f90a610a80e564622eeec3d007a151383c69b2b1aR165